### PR TITLE
CP-46155: Call SM scripts via observer.py when SM observer is enabled

### PR DESCRIPTION
**Pull request by snwoods**

### CODE REVIEW

**Comment by snwoods** [comment]
This has passed a ring3 BVT suiterun 194264 and a Storage BVT suiterun 194368

**Review by mg12 — APPROVED** [review]

**Review by GabrielBuica — APPROVED** [review]
LGTM! :)

File: ocaml/xapi/xapi_observer_components.ml [hunk 1]
```diff
@@ -84,18 +84,26 @@ let ( // ) = Filename.concat
...
-        @
-        match traceparent with
-        | None ->
-            []
-        | Some traceparent ->
-            [Env_record.pair ("TRACEPARENT", traceparent)]
-        )
-    ]
+  let env_vars =
+    Array.concat
```
Comment (lindig) [inline ocaml/xapi/xapi_observer_components.ml:90]: This looks complicated. Could this be simplified using the fact that `[| |]` is a neutral element `Array.concat`?

```
let a = Array.concat
	[ [|"foo";"bar"|]
	; match x with
	  | Some _ -> [| "baz" |]
		| None -> [| |]
	]
```

**Review by lindig — APPROVED** [review]

File: ocaml/xapi/xapi_observer_components.ml (hunk 1 above)
Comment (Vincent-lau) [reply ocaml/xapi/xapi_observer_components.ml:90]: Very minor thing, but IMO `Option.fold` might be a bit more reable

**Pull request merged**